### PR TITLE
feat: Menu component

### PR DESCRIPTION
**Pull request by JiLiZART**

### CODE REVIEW

File: src/components/pickers/Menu/Menu.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  MenuTrigger,
+  Flex,
+  Button,
+  Text,
+  Root,
+  Space,
+} from '../../../index';
+import { baseProps } from '../../../stories/lists/baseProps';
+
+const Layout = ({ children }) => <Root>{children}</Root>;
```
Comment (tenphi) [inline src/components/pickers/Menu/Menu.stories.tsx:14]: We have the default wrapper with Root. So the additional wrapper should not be required.

File: src/components/pickers/Menu/Menu.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    ...mergeProps(contextProps, props, {
+      mods: {
+        footer: !!footer,
+        header: !!header,
+      },
+    }),
+  };
+  const state = useTreeState(completeProps);
+  const { menuProps } = useMenu(completeProps, state, domRef);
+  const styleProps = extractStyles(completeProps, CONTAINER_STYLES);
```
Comment (tenphi) [inline src/components/pickers/Menu/Menu.tsx:47]: `extractStyles` extracts styles, not props.

File: src/components/pickers/Menu/Menu.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+const _Menu = React.forwardRef(Menu) as <T>(
+  props: CubeMenuProps<T> & { ref?: DOMRef<HTMLUListElement> },
+) => ReactElement;
+
+type PartialMenuButton = Partial<MenuButtonProps>;
+
+type ItemComponent = <T>(
+  props: ItemProps<T> & PartialMenuButton,
+) => JSX.Element;
+type SectionComponent = <T>(props: SectionProps<T>) => JSX.Element;
```
Comment (tenphi) [inline src/components/pickers/Menu/Menu.tsx:100]: Too complicated and can cause an error. We can replace it with a simple `typeof Section`.

File: src/components/pickers/Menu/MenuButton.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    hovered: '#dark.04',
+    pressed: '#purple.1',
+    selected: '#purple.1',
+    disabled: '#clear',
+  },
+  color: {
+    '': '#dark-02',
+    hovered: '#dark-02',
+    pressed: '#purple-text',
+    selected: '#purple-text',
```
Comment (tenphi) [inline src/components/pickers/Menu/MenuButton.tsx:24]: Can simplified too.

File: src/components/pickers/Menu/MenuItem.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          disabled: isDisabled,
+          selected: isSelected,
+          selectable: isSelectable,
+          hovered: isHovered,
+        }}
+      >
+        <ClearSlots>
+          <SlotProvider
+            slots={{
+              text: { className: 'itemLabel', ...labelProps },
```
Comment (tenphi) [inline src/components/pickers/Menu/MenuItem.tsx:76]: We do not use classes. We can replace it with `data-qa="MenuItem_Label". The same goes for the slots below.

File: src/components/pickers/Menu/styled.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      '': '0.5x',
+      section: '0',
+    },
+    border: {
+      '': '#dark.05',
+      section: '',
+    },
+    radius: {
+      '': '1r',
+      header: '1x 1x 1r 1r',
```
Comment (tenphi) [inline src/components/pickers/Menu/styled.tsx:22]: Weird mix of `1x` and `1r`.

File: src/components/pickers/Menu/styled.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    boxShadow: {
+      '': '',
+      popover: '0px 5px 15px rgba(var(--dark-color-rgb), 0.05)',
```
Comment (tenphi) [inline src/components/pickers/Menu/styled.tsx:26]: `#dark.05` instead of `rgba...`.

File: src/components/pickers/Menu/styled.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+});
+
+export const StyledMenuHeader = styled(Space, {
+  tag: 'li',
+  styles: {
+    fill: '#light-02',
+    color: '#dark-02',
+    padding: '0.75x 2x',
+    margin: '-0.5x -0.5x (0.5x - 1px) -0.5x',
+    borderBottom: '#dark.04',
```
Comment (tenphi) [inline src/components/pickers/Menu/styled.tsx:50]: can be replaced with `border: '#dark.04 bottom'`.

File: package.json [hunk 9]
```diff
@@ -63,11 +63,13 @@
     "@react-aria/i18n": "^3.3.4",
     "@react-aria/interactions": "^3.7.0",
     "@react-aria/listbox": "^3.4.0",
+    "@react-aria/menu": "3.4.3",
     "@react-aria/numberfield": "^3.1.1",
     "@react-aria/overlays": "^3.7.3",
     "@react-aria/radio": "^3.1.6",
     "@react-aria/searchfield": "^3.2.2",
     "@react-aria/select": "^3.6.0",
+    "@react-aria/separator": "3.1.6",
```
Comment (tenphi) [inline package.json:72]: Looks like we don't use it. Maybe we can add it to the Divider component and add an example with it and Menu to the Storybook.

File: src/components/pickers/Menu/styled.tsx [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    placeContent: 'space-between',
+    align: 'start',
+    fontWeight: '500',
```
[the remark this answers is not in the document]
Comment (JiLiZART) [reply src/components/pickers/Menu/styled.tsx:53]: Where I can find presets for fonts?

File: src/tokens.ts [hunk 11]
```diff
@@ -16,6 +16,7 @@ const colors = {
   'dark-05': '213, 213, 226',
   'grey-light': '248, 248, 249',
   light: '246, 246, 248',
+  'light-02': '249, 249, 253',
```
[the remark this answers is not in the document]
Comment (JiLiZART) [reply src/tokens.ts:19]: Yep

File: src/components/pickers/Menu/styled.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (tenphi) [reply src/components/pickers/Menu/styled.tsx]: `tokens.ts` starting at the 85th line.

File: src/components/pickers/Menu/styled.tsx (hunk 7 above)
Comment (JiLiZART) [reply src/components/pickers/Menu/styled.tsx:25]: `0px 5px 15px #dark.05` ?

File: package.json (hunk 9 above)
Comment (JiLiZART) [reply package.json:72]: I think it' for separate PR

**Review by tenphi — APPROVED** [review]

**Pull request merged**